forge/routes/api/deviceLive.js

```diff
@@ -74,6 +84,36 @@ module.exports = async function (app) {
 
     app.get('/snapshot', async (request, reply) => {
         if (!request.device.targetSnapshot) {
+            // determine is device is in application mode? if so, return a default snapshot to permit the user to generate flows
+            const device = request.device || null
+            const ownerIsApplication = device?.ownerType === 'application' // && 'EE'?
+            if (ownerIsApplication) {
+                const DEFAULT_APP_SNAPSHOT = {
+                    id: '0',
+                    name: 'Starter Snapshot',
+                    description: 'A starter snapshot for new applications',
+                    flows: [
+                        { id: 'FFF0000000000001', type: 'tab', label: 'FlowFuse Device Flow', disabled: false, info: '' },
+                        { id: 'FFCOM00000000001', type: 'comment', z: 'FFF0000000000001', name: 'Welcome to Node-RED by FlowFuse! \\n This is a basic starter flow for your new device, to get you started.', info: '', x: 310, y: 80 },
+                        { id: 'FFINJ00000000001', type: 'inject', z: 'FFF0000000000001', name: 'On Start', props: [{ p: 'payload' }, { p: 'topic', vt: 'str' }], repeat: '', crontab: '', once: true, onceDelay: '0.2', topic: '', payload: 'true', payloadType: 'bool', x: 140, y: 160, wires: [['FFCHA00000000001']] },
+                        { id: 'FFCHA00000000001', type: 'change', z: 'FFF0000000000001', name: 'Get Env Vars', rules: [{ t: 'set', p: 'payload', pt: 'msg', to: '{}', tot: 'json' }, { t: 'set', p: 'payload.device', pt: 'msg', to: 'FF_DEVICE_NAME', tot: 'env' }, { t: 'set', p: 'payload.application', pt: 'msg', to: 'FF_APPLICATION_NAME', tot: 'env' }], action: '', reg: false, x: 320, y: 160, wires: [['FFDBG00000000001']] },
+                        { id: 'FFDBG00000000001', type: 'debug', z: 'FFF0000000000001', name: 'Info', active: true, tosidebar: true, console: true, tostatus: true, complete: 'payload', targetType: 'msg', statusVal: 'payload', statusType: 'auto', x: 490, y: 160 }
+                    ],
+                    modules: {
+                        'node-red': 'latest' // TODO: get this from the "somewhere!?!?" - this is where TAGs might work well.
+                    },
+                    env: {
+                        FF_SNAPSHOT_ID: '0',
+                        FF_SNAPSHOT_NAME: 'None',
```

Comment:
```suggestion
                        FF_SNAPSHOT_ID: '0',
                        FF_SNAPSHOT_NAME: 'None',
```

Same comment as elsewhere - I wonder if, when exposing in the flow, these should be blank rather than placeholders. Not sure.